CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+cmake_minimum_required(VERSION 3.18.0)
+project(otter DESCRIPTION "use this cmake to generate compiler db JSON file")
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_INSTALL_MESSAGE LAZY CACHE STRING "" FORCE)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON CACHE BOOL "" FORCE)
+
+# Find ERTS
+find_program(ELIXIR_EXECUTABLE NAMES elixir REQUIRED)
+execute_process(
+  COMMAND ${ELIXIR_EXECUTABLE} --eval ":code.root_dir() |> IO.puts()"
+  OUTPUT_VARIABLE ERTS_DIR
+  COMMAND_ERROR_IS_FATAL ANY
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+execute_process(
+  COMMAND ${ELIXIR_EXECUTABLE} --eval ":erlang.system_info(:version) |> IO.puts()"
+  OUTPUT_VARIABLE ERTS_VERSION
+  COMMAND_ERROR_IS_FATAL ANY
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+execute_process(
+  COMMAND ${ELIXIR_EXECUTABLE} --eval ":code.lib_dir(:erl_interface, :include) |> IO.puts()"
+  OUTPUT_VARIABLE ERL_INTERFACE_INCLUDE_DIR
+  COMMAND_ERROR_IS_FATAL ANY
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+execute_process(
+  COMMAND ${ELIXIR_EXECUTABLE} --eval ":code.lib_dir(:erl_interface, :lib) |> IO.puts()"
+  OUTPUT_VARIABLE ERL_INTERFACE_LIB_DIR
+  COMMAND_ERROR_IS_FATAL ANY
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
```

Comment:
Instead of calling `elixir` on the commandline, can you use the corresponding environment variables from `elixir_make`. See https://hexdocs.pm/elixir_make/Mix.Tasks.Compile.ElixirMake.html#module-default-environment-variables. This will make compilation work with Nerves and it should be a little faster.

Reply:
At this point, this cmake serves the purpose of generating `compile_commands.json` file for Clangd/CLion and other things alike. Mix will not use it instead it still uses the top level makefile to build. 
I guess if will make things easier if we want to build libs for testing if we switch to cmake?
@cocoa-xu

Reply:
If `CMakeLists.txt` and `compile_commands.json` are not used by this project, then perhaps just leave them in your local env?

Reply:
I mean, this could cause some confusion as people see there is a `CMakeLists.txt` file, then naturally would assume this project depends on `cmake`.

Reply:
Agree

Reply:
I was thinking someone else contributing to the project might need it as well

Reply:
Also, if later for whatever reason we have to use cmake for this project (instead of for clangd/clion), then it will definitely cause problems for clangd/clion.

Reply:
I see. Thanks for the explanation.

Reply:
With that said, I totally agree that autocomplete and other features provided by CLion or other IDEs indeed make it easier to write NIFs.

Reply:
removed